backend/src/main/java/ai/verta/modeldb/versioning/RepositoryDAORdbImpl.java

```diff
@@ -389,40 +366,16 @@ private void deleteRoleBindingsOfAccessibleResources(List<RepositoryEntity> allo
             Code.PERMISSION_DENIED);
       }
 
-      ListBranchesRequest.Response listBranchesResponse =
-          listBranches(
-              ListBranchesRequest.newBuilder().setRepositoryId(request.getRepositoryId()).build());
-      if (!listBranchesResponse.getBranchesList().isEmpty()) {
-        String deleteBranchesHQL =
-            "DELETE FROM "
-                + BranchEntity.class.getSimpleName()
-                + " br where br.id.repository_id = :repositoryId AND br.id.branch IN (:branches)";
-        Query deleteBranchQuery = session.createQuery(deleteBranchesHQL);
-        deleteBranchQuery.setParameter("repositoryId", repository.getId());
-        deleteBranchQuery.setParameterList("branches", listBranchesResponse.getBranchesList());
-        deleteBranchQuery.executeUpdate();
-      }
-
-      CommitPaginationDTO commitPaginationDTO =
-          commitDAO.fetchCommitEntityList(
-              session, ListCommitsRequest.newBuilder().build(), repository.getId());
-      commitPaginationDTO
-          .getCommitEntities()
-          .forEach(
-              commitEntity -> {
-                if (commitEntity.getRepository().contains(repository)) {
-                  commitEntity.getRepository().remove(repository);
-                  if (commitEntity.getRepository().isEmpty()) {
-                    session.delete(commitEntity);
-                  } else {
-                    session.update(commitEntity);
-                  }
-                }
-              });
+      Query deletedRepositoriesQuery =
+          session.createQuery(UPDATE_DELETED_STATUS_REPOSITORY_QUERY_STRING);
+      deletedRepositoriesQuery.setParameter("deleted", true);
+      deletedRepositoriesQuery.setParameter("repoIds", allowedRepositoryIds);
+      int updatedCount = deletedRepositoriesQuery.executeUpdate();
+      LOGGER.debug(
+          "Mark Repositories as deleted : {}, count : {}", allowedRepositoryIds, updatedCount);
       // Delete all VersionedInputs for repository ID
       experimentRunDAO.deleteLogVersionedInputs(session, repository.getId(), null);
-
-      deleteRoleBindingsOfAccessibleResources(Collections.singletonList(repository));
+      transaction.commit();
       session.delete(repository);
       session.getTransaction().commit();
```

Comment:
Looks like this is committing the transaction twice, right?

Reply:
Yes, Fixed